inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/scheduling/tasks/TrainingTask.java

```diff
@@ -133,17 +142,28 @@ public void run()
         recommendationScheduler.enqueue(new PredictionTask(user, getProject()));
     }
 
-    private List<CAS> readCasses(Project aProject, User aUser)
+    private List<AnnotationCas> readCasses(Project aProject, User aUser)
     {
-        List<CAS> casses = new ArrayList<>();
+        List<AnnotationCas> casses = new ArrayList<>();
         for (AnnotationDocument doc : documentService.listAnnotationDocuments(aProject, aUser)) {
             try {
                 JCas jCas = documentService.readAnnotationCas(doc);
-                casses.add(jCas.getCas());
+                casses.add(new AnnotationCas(jCas.getCas(), doc.getState()));
             } catch (IOException e) {
                 log.error("Cannot read annotation CAS.", e);
             }
         }
         return casses;
     }
+
+    private static class AnnotationCas
```

Comment:
I find the name confusing. How about `TrainingDocument`?

Reply:
I am thinking right now that it would be easier to store the states we want to ignore instead of the states we want to use for training. The changeset is not so trivial to do.

Reply:
We could implement a custom task in Java which does what we want. I'm not sure if reversing the logic would bite us back later at some point. Although in the worst case, we could try it with the reversed logic and implement a custom task to again reverse it later if it bites us...

Reply:
Then I implement the task for the non-reversed logic.

Reply:
I reversed the logic.